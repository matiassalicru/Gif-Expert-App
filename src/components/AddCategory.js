import React, { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState("");
  const [type, settype] = useState(false)

  const handleInputChange = (e) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    console.log('HandleSubmit');
    console.log(inputValue);

    if (inputValue.length < 2) {
      settype(true)
      handleError();
    } else {
      if (inputValue.trim().length >= 2) {
        setCategories(cats=> [inputValue, ...cats]);
        setInputValue("");
      }
    }
    
  };

  const handleError = () =>{

    setTimeout(() => {
      settype(false);
    }, 2000);

  }

  return (
    <form onSubmit={handleSubmit}>
      <p>{inputValue}</p>
      <input
        placeholder="Search"
        type="text"
        value={inputValue}
        onChange={handleInputChange}
      />

      { type &&
        <div id="error">Please type more than one character</div>
      }

    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
