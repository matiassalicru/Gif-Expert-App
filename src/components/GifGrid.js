import React from "react";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifGridItem } from "./GifGridItem";
import PropTypes from "prop-types";

export const GifGrid = ({ category }) => {
  const { data: images, loading } = useFetchGifs(category);

  return (
    <>
      <h3 className="category">{category}</h3>

      {loading && <p className=" loading animate__animated animate__flip">Loading</p>}

      <div className="card-grid animate__animated animate__backInRight">
        {images.map((img) => (
          <GifGridItem key={img.id} {...img} />
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
