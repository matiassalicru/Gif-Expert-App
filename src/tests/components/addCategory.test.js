import React from "react";
import { shallow } from "enzyme";
import { AddCategory } from "../../components/AddCategory";

describe("Pruebas en el componente", () => {
  const setCategories = () => {};
  const wrapper = shallow(<AddCategory setCategories={setCategories} />);

  test("Debe de mostrarse correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de cambiar la caja de texto, el input", () => {
    const input = wrapper.find("input");
    const value = "";
    input.simulate("change", {
      target: {
        value,
      },
    });

    expect(wrapper.find("p").text().trim()).toBe(value);
  });
});
