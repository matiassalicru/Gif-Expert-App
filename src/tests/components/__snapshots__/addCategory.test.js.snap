// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente Debe de mostrarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    placeholder="Search"
    type="text"
    value=""
  />
  <div
    id="error"
  />
</form>
`;
